src/SdfGenerator.cc

```diff
@@ -422,6 +430,67 @@ namespace sdf_generator
     return true;
   }
 
+  /////////////////////////////////////////////////
+  /// \brief Checks if a string is a number
+  /// \param[in] _str The string to check
+  /// \return True if the string is a number
+  bool isNumber(const std::string &_str)
+  {
+    for (const char &c : _str)
+      if (!std::isdigit(c)) return false;
+
+    return true;
+  }
+
+  /////////////////////////////////////////////////
+  void updateModelElementWithNestedInclude(sdf::ElementPtr &_elem,
+                                           const bool _saveFuelVersion,
+                                           const IncludeUriMap &_includeUriMap)
+  {
+    sdf::ElementPtr e = _elem->GetFirstElement(), nextE = nullptr;
+    while (e != nullptr)
+    {
+      nextE = e->GetNextElement();
+
+      if (e->GetIncludeElement() != nullptr)
+      {
+        std::string modelDir = common::parentPath(e->FilePath());
+        auto uriMapIt = _includeUriMap.find(modelDir);
+
+        if (_saveFuelVersion && uriMapIt != _includeUriMap.end())
+        {
+          // find fuel model version from file path
+          std::string version = common::parentPath(e->FilePath());
+          version = common::basename(version);
+
+          if (isNumber(version))
+          {
+            sdf::ElementPtr uriElem = e->GetIncludeElement()->GetElement("uri");
+            std::string uri = uriElem->GetValue()->GetAsString();
```

Comment:
tip: This can be simplified as `e->GetIncludeElement()->Get<std::string>("uri")`

Reply:
73b71101702286e5161854ec650ff9aa39c0ed0d